frontend/src/metabase/actions/containers/ActionCreator/FormCreator/OptionEditor.tsx

```diff
@@ -5,29 +5,72 @@ import TippyPopoverWithTrigger from "metabase/components/PopoverWithTrigger/Tipp
 import Button from "metabase/core/components/Button";
 import Icon from "metabase/components/Icon";
 
+import type { FieldType } from "metabase-types/api";
+
 import {
   OptionEditorContainer,
   AddMorePrompt,
+  ErrorMessage,
   TextArea,
 } from "./OptionEditor.styled";
 
-type ValueOptions = (string | number)[];
+export type ValueOptions = (string | number)[];
 
 const optionsToText = (options: ValueOptions) => options.join("\n");
 const textToOptions = (text: string): ValueOptions =>
   text.split("\n").map(option => option.trim());
 
+function cleanOptions(options: ValueOptions, fieldType: FieldType) {
+  if (fieldType === "number") {
+    return options.map(option => Number(option));
+  }
+  return options;
+}
+
+function validateOptions(options: ValueOptions, fieldType: FieldType) {
+  if (fieldType === "number") {
+    const isValid = options.every(option => !Number.isNaN(option));
+    return isValid ? undefined : t`Invalid number format`;
+  }
+  return;
+}
+
+export interface OptionEditorProps {
+  fieldType: FieldType;
+  options: ValueOptions;
+  onChange: (options: ValueOptions) => void;
+}
+
 export const OptionPopover = ({
+  fieldType,
   options,
   onChange,
-}: {
-  options: ValueOptions;
-  onChange: (options: ValueOptions) => void;
-}) => {
+}: OptionEditorProps) => {
   const [text, setText] = useState(optionsToText(options));
-  const save = (closePopover: () => void) => {
-    onChange(textToOptions(text));
-    closePopover();
+  const [error, setError] = useState<string | null>(null);
+
+  const hasOptions = text.length > 0;
+  const isDirty = text !== optionsToText(options);
+  const hasError = Boolean(error);
+  const canSave = hasOptions && isDirty && !hasError;
+
+  const handleTextChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setText(e.target.value);
+    if (hasError) {
+      setError(null);
+    }
+  };
+
+  const handleSave = (closePopover: () => void) => {
+    setError(null);
+    const nextOptions = cleanOptions(textToOptions(text), fieldType);
+    const error = validateOptions(nextOptions, fieldType);
+    if (error) {
+      setError(error);
+    } else {
```

Comment:
This is great. ❤️ 

![Screen Shot 2023-03-07 at 3 12 19 PM](https://user-images.githubusercontent.com/30528226/223565518-f3755de6-9726-47a0-b738-b265a3376a3a.png)